### PR TITLE
MGDOBR-839: implement ansible_tower_job_template_sink_0.1 action

### PR DESCRIPTION
**Pull request by kostola**
[MGDOBR-839](https://issues.redhat.com/browse/MGDOBR-839)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] The layers in the `kustomize` folder have been updated accordingly.
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

**Review by r00ta — APPROVED** [review]

**Comment by r00ta** [comment]
There are a couple of tests that you have to update here
https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/118660a0fe3a735a97fed78468b0b4948a403aa8/manager/src/test/java/com/redhat/service/smartevents/manager/api/user/SchemaAPITest.java#L31

and here 
https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/118660a0fe3a735a97fed78468b0b4948a403aa8/manager/src/test/java/com/redhat/service/smartevents/manager/api/user/SchemaAPITest.java#L57

**Comment by r00ta** [comment]
Once you merge this, I'll rebase https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/834 and move the custom validations to the json schema

File: processor/configuration/src/main/java/com/redhat/service/smartevents/processor/resolvers/custom/AnsibleTowerJobTemplateActionResolver.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        transformedAction.setType(WebhookAction.TYPE);
+        transformedAction.setMapParameters(parameters);
+
+        return transformedAction;
+    }
+
+    static String getLaunchTemplateEndpoint(Action action) throws MalformedURLException {
+        String baseEndpoint = action.getParameter(ENDPOINT_PARAM);
+        String jobTemplateId = action.getParameter(JOB_TEMPLATE_ID_PARAM);
+        String fullEndpoint = String.format("%s/api/v2/job_templates/%s/launch/", baseEndpoint, jobTemplateId);
```
Comment (danielezonca) [inline processor/configuration/src/main/java/com/redhat/service/smartevents/processor/resolvers/custom/AnsibleTowerJobTemplateActionResolver.java:49]: Move string to constant

File: processor/configuration/src/main/java/com/redhat/service/smartevents/processor/resolvers/custom/AnsibleTowerJobTemplateActionResolver.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        }
+        if (action.hasParameter(SSL_VERIFICATION_DISABLED)) {
+            parameters.put(SSL_VERIFICATION_DISABLED, action.getParameter(SSL_VERIFICATION_DISABLED));
+        }
+
+        Action transformedAction = new Action();
+        transformedAction.setType(WebhookAction.TYPE);
+        transformedAction.setMapParameters(parameters);
+
+        return transformedAction;
```
Comment (danielezonca) [inline processor/configuration/src/main/java/com/redhat/service/smartevents/processor/resolvers/custom/AnsibleTowerJobTemplateActionResolver.java:43]: Side comment, what about the payload? Is it the whole message? Checking Ansible Tower documentation it seems the payload to be an arbitrary JSON so the whole payload should work
https://docs.ansible.com/ansible-tower/latest/html/towerapi/api_ref.html#/

**Review by danielezonca — APPROVED** [review]
Minor comments but LGTM

**Review by jiripetrlik — COMMENTED** [review]
I have one minor comment.

File: processor/configuration/src/main/java/com/redhat/service/smartevents/processor/resolvers/custom/AnsibleTowerJobTemplateActionResolver.java (hunk 2 above)
Comment (kostola) [reply processor/configuration/src/main/java/com/redhat/service/smartevents/processor/resolvers/custom/AnsibleTowerJobTemplateActionResolver.java:43]: The payload is the expected payload from Ansible Tower API call, it's up to the user to transform the original event into a compatible payload. It must be documented in the action reference.

File: processor/configuration/src/main/java/com/redhat/service/smartevents/processor/resolvers/custom/AnsibleTowerJobTemplateActionResolver.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+public class AnsibleTowerJobTemplateActionResolver implements AnsibleTowerJobTemplateAction, GatewayResolver<Action> {
+
+    @Override
+    public Action resolve(Action action, String customerId, String bridgeId, String processorId) {
+        Map<String, String> parameters = new HashMap<>();
+        try {
+            String launchTemplateEndpoint = getLaunchTemplateEndpoint(action);
+            parameters.put(WebhookAction.ENDPOINT_PARAM, launchTemplateEndpoint);
+        } catch (MalformedURLException e) {
+            throw new GatewayProviderException("Can't build API call for ansible tower instance");
```
[the remark this answers is not in the document]
Comment (kostola) [reply processor/configuration/src/main/java/com/redhat/service/smartevents/processor/resolvers/custom/AnsibleTowerJobTemplateActionResolver.java]: Thanks for the suggestion. Changed :+1:

File: processor/configuration/src/main/java/com/redhat/service/smartevents/processor/resolvers/custom/AnsibleTowerJobTemplateActionResolver.java (hunk 1 above)
Comment (kostola) [reply processor/configuration/src/main/java/com/redhat/service/smartevents/processor/resolvers/custom/AnsibleTowerJobTemplateActionResolver.java]: Done :+1:

**Review by jiripetrlik — APPROVED** [review]

**Pull request merged**